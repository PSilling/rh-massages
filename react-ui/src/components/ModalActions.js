// react imports
import React, { Component } from 'react';

// util imports
import _t from '../utils/Translations.js';

/**
 * Custom component rendering action buttons for modal footers.
 */
class ModalActions extends Component {

  render() {
    return(
      <div className="pull-right">
        <button type="button" className="btn btn-primary" onClick={this.props.onProceed}
          style={{ 'marginRight': '5px' }} autoFocus={this.props.autoFocus}>
          {this.props.primaryLabel}
        </button>
        <button type="button" className="btn btn-default"
          onClick={this.props.onClose}>
          { _t.translate('Dismiss') }
        </button>
      </div>
    );
  }
}

export default ModalActions
