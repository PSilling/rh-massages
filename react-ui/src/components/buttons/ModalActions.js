// react imports
import React from "react";
import PropTypes from "prop-types";

// module imports
import { Button, Col, Row } from "reactstrap";

// util imports
import _t from "../../util/Translations";

/**
 * Footer buttons for Modal dialogs. Children components can be given to add more
 * case specific elements before the two main buttons.
 */
const ModalActions = function ModalActions(props) {
  return (
    <Row className="text-right">
      <Col md="12">
        <hr />
        {props.children}
        {props.primaryLabel !== "none" && (
          <Button className="mr-2" color="primary" onClick={props.onProceed} disabled={props.disabled}>
            {props.primaryLabel}
          </Button>
        )}
        <Button onClick={props.onClose}>{_t.translate("Dismiss")}</Button>
      </Col>
    </Row>
  );
};

ModalActions.propTypes = {
  /** callback function triggered on close button click */
  onClose: PropTypes.func.isRequired,
  /** callback function triggered on primary button click */
  onProceed: PropTypes.func.isRequired,
  /** extra elements to be provided before the two buttons */
  children: PropTypes.oneOfType([PropTypes.arrayOf(PropTypes.node), PropTypes.node]),
  /** whether the primary button should be disabled */
  disabled: PropTypes.bool,
  /** primary button label to be displayed */
  primaryLabel: PropTypes.string
};

ModalActions.defaultProps = {
  children: null,
  disabled: false,
  primaryLabel: _t.translate("Proceed")
};

export default ModalActions;
