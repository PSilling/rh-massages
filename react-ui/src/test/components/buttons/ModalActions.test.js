// react imports
import React from "react";
import TestRenderer from "react-test-renderer";

// test imports
import { Button } from "reactstrap";
import ModalActions from "../../../components/buttons/ModalActions";
import _t from "../../../util/Translations";

test("renders content with correct props", () => {
  const testFunctionProceed = jest.fn();
  const testFunctionClose = jest.fn();
  const testRenderer = TestRenderer.create(
    <ModalActions onProceed={testFunctionProceed} onClose={testFunctionClose}>
      <button type="button" onClick={testFunctionProceed} />
    </ModalActions>
  );
  const testInstance = testRenderer.root;
  const treeJSON = testRenderer.toJSON();
  const buttons = testInstance.findAllByType(Button);
  const standardButtons = testInstance.findAllByType("button");

  expect(testFunctionProceed).not.toHaveBeenCalled();

  buttons[0].props.onClick();

  expect(testFunctionProceed).toHaveBeenCalledTimes(1);
  expect(testFunctionClose).not.toHaveBeenCalled();

  buttons[1].props.onClick();

  expect(testFunctionClose).toHaveBeenCalledTimes(1);
  expect(buttons[0].props.children).toEqual(_t.translate("Proceed"));
  expect(buttons[0].props.onClick).toBe(testFunctionProceed);
  expect(buttons[0].props.disabled).toBe(false);
  expect(buttons[1].props.children).toBe(_t.translate("Dismiss"));
  expect(buttons[1].props.onClick).toBe(testFunctionClose);
  expect(standardButtons[0].props.onClick).toBe(testFunctionProceed);
  expect(treeJSON).toMatchSnapshot();
});

test("hides primary button when primaryLabel equals none", () => {
  const testFunctionProceed = jest.fn();
  const testFunctionClose = jest.fn();
  const testRenderer = TestRenderer.create(
    <ModalActions primaryLabel="none" onProceed={testFunctionProceed} onClose={testFunctionClose} />
  );
  const testInstance = testRenderer.root;
  const button = testInstance.findByType(Button);

  expect(button.props.onClick).toBe(testFunctionClose);
});
