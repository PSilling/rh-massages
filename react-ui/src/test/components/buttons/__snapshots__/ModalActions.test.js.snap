// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders content with correct props 1`] = `
<div
  className="text-right row"
>
  <div
    className="col-md-12"
  >
    <hr />
    <button
      onClick={[Function]}
      type="button"
    />
    <button
      aria-label={null}
      className="mr-2 btn btn-primary"
      disabled={false}
      onClick={[Function]}
      type="button"
    >
      Pokračovat
    </button>
    <button
      aria-label={null}
      className="btn btn-secondary"
      onClick={[Function]}
      type="button"
    >
      Zrušit
    </button>
  </div>
</div>
`;
